weave/monitoring/openai/openai.py

```diff
@@ -84,6 +84,41 @@ async def __aiter__(self) -> AsyncIterator[ChatCompletionChunk]:
         self._finish_run(result_with_usage.model_dump(exclude_unset=True))
 
 
+class WeaveStream(Stream):
```

Comment:
New class - pretty simple. It follows the same pattern as the async class above and uses the same code that was previously in a function below.